src/Microsoft.AspNetCore.Blazor/Builder/BlazorApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Blazor.Components;
+
+namespace Microsoft.AspNetCore.Blazor.Builder
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="IBlazorApplicationBuilder"/>.
+    /// </summary>
+    public static class BlazorApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Associates the component type with the application,
+        /// causing it to be displayed in the specified DOM element.
+        /// </summary>
+        /// <param name="app">The <see cref="IBlazorApplicationBuilder"/>.</param>
+        /// <typeparam name="TComponent">The type of the component.</typeparam>
+        /// <param name="domElementSelector">A CSS selector that uniquely identifies a DOM element.</param>
+        public static void AddComponent<TComponent>(this IBlazorApplicationBuilder app, string domElementSelector)
+            where TComponent : IComponent
+        {
+            app.AddComponent(typeof(TComponent), domElementSelector);
+        }
```

Comment:
We didn't get to a super high level of clarity about this earlier. I'm happy to move this to the browser library for now and we can revisit when/if it becomes necessary elsewhere

Reply:
Yes, it would be good to have it in the `.Browser` project because of it having DOM-specific concepts.

I hopeful the layering will pay off with some extra clarity when we get to server-side prerendering and unit test helpers, but that remains to be seen. We can merge the browser/nonbrowser layers together later if it turns out the distinction isn't helping.